backend/src/main/java/com/bakdata/conquery/models/events/BucketManager.java

```diff
@@ -231,78 +204,46 @@ private void deregisterBucket(Bucket bucket) {
 			}
 		}
 
-		tableToBuckets.getOrDefault(bucket.getTable().getId(), Int2ObjectMaps.emptyMap())
+		tableToBuckets.getOrDefault(bucket.getTable(), Int2ObjectMaps.emptyMap())
 					  .getOrDefault(bucket.getBucket(), Collections.emptyList())
 					  .removeIf(bkt -> bkt.getId().equals(bucket.getId()));
 
 	}
 
-	public void removeBucket(BucketId bucketId) {
-		Bucket bucket = storage.getBucket(bucketId);
-		if (bucket == null) {
-			return;
-		}
-
-		for (Concept<?> concept : storage.getAllConcepts()) {
-			for (Connector con : concept.getConnectors()) {
-				try (Locked lock = cBlockLocks.acquire(con.getId())) {
-					removeCBlock(new CBlockId(bucketId, con.getId()));
-				}
-			}
-		}
+	public void removeBucket(Bucket bucket) {
+		storage.getAllCBlocks()
+			   .stream()
+			   .filter(cblock -> cblock.getBucket().equals(bucket))
+			   .forEach(this::removeCBlock);
 
 		deregisterBucket(bucket);
 
-		storage.removeBucket(bucketId);
+		storage.removeBucket(bucket.getId());
 	}
 
-	public void removeConcept(ConceptId conceptId) {
-		Concept<?> c = storage.getConcept(conceptId);
-
-		if (c == null) {
-			return;
-		}
-
-		for (Connector con : c.getConnectors()) {
-			for (Import imp : con.getTable().findImports(storage)) {
-				for (int bucketNumber : worker.getInfo().getIncludedBuckets()) {
+	public void removeConcept(Concept<?> concept) {
 
-					BucketId bucketId = new BucketId(imp.getId(), bucketNumber);
+		storage.getAllCBlocks().stream()
+			   .filter(cBlock -> cBlock.getConnector().getConcept().equals(concept))
+			   .forEach(this::removeCBlock);
 
-					if (storage.getBucket(bucketId) == null) {
-						log.trace("Skipping cleanup of CBlock of Concept[{}] for Bucket[{}], because it is not present", conceptId, bucketId);
-						continue;
-					}
-
-					removeCBlock(new CBlockId(bucketId, con.getId()));
-				}
-			}
-		}
-
-		storage.removeConcept(conceptId);
+		storage.removeConcept(concept.getId());
```

Comment:
pew das ist auch schon ganz schon verschachtelt. Wird hier `connectorToCblocks`  von dem Connector bereinigt?

Reply:
jetzt ist es nicht mehr so verschachtelt wie davor. Die werden in deregisterBukcket/deregisterCBlock gepflegt. ich hab die noch etwas aufgeräumt.

Reply:
@awildturtok Ah das war der Kommentar